### PR TITLE
Fixing XSS vulnerabilities in userprofile ui.

### PR DESCRIPTION
**Pull request by ChamaraPhilipsuom**

### CODE REVIEW

**Comment by ChamaraPhilipsuom** [comment]
Fixed XSS issues in userprofile.

File: components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/add.jsp [hunk 1]
```diff
@@ -148,7 +148,8 @@ function validateTextForIllegal(fld,fldName) {
 
                 <% if (userFields[i].getRequired()&& userFields[i].getDisplayName()!=null) {%>
                     if (validateEmpty("<%=userFields[i].getClaimUri()%>").length > 0) {
-                        CARBON.showWarningDialog("<%=userFields[i].getDisplayName()%>" + " <fmt:message key='is.required'/>");
+                        CARBON.showWarningDialog("<%=Encode.forJavaScriptAttribute(userFields[i].getDisplayName())%>" +
```
Comment (malithie) [inline components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/add.jsp]: encode for html and then encode for javascript

File: components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/add.jsp [hunk 2]
```diff
@@ -158,7 +159,8 @@ function validateTextForIllegal(fld,fldName) {
                     var reg = new RegExp("<%=userFields[i].getRegEx() %>");
```
Comment (malithie) [inline components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/add.jsp]: This line should be encoded for javascript

File: components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/add.jsp [hunk 3]
```diff
@@ -158,7 +159,8 @@ function validateTextForIllegal(fld,fldName) {
...
                     var valid = reg.test(value);
                     if (value != '' && !valid) {
-                        CARBON.showWarningDialog("<%=userFields[i].getDisplayName()%>" + " <fmt:message key='is.not.valid'/>");
+                        CARBON.showWarningDialog("<%=Encode.forJavaScriptAttribute(userFields[i].getDisplayName())%>" +
```
Comment (malithie) [inline components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/add.jsp]: encode for html and then encode for javascript

File: components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/edit.jsp [hunk 4]
```diff
@@ -85,7 +85,7 @@
 %>
 <script type="text/javascript">
     function forward() {
-        location.href = "<%=forwardTo%>";
+        location.href = "<%=Encode.forJavaScriptBlock(forwardTo)%>";
```
Comment (malithie) [inline components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/edit.jsp]: forwardTo is hardcoded and does not contain untrusted data. Not necessary to encode

File: components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/edit.jsp [hunk 5]
```diff
@@ -155,7 +155,8 @@
                         var value = document.getElementsByName("<%=userFields[i].getClaimUri()%>")[0].value;
                         <%if (userFields[i].getRequired() && userFields[i].getDisplayName()!=null) {%>
                             if (validateEmpty("<%=userFields[i].getClaimUri()%>").length > 0) {
-                                CARBON.showWarningDialog("<%=CharacterEncoder.getSafeText(userFields[i].getDisplayName())%>" + " <fmt:message key='is.required'/>");
+                                CARBON.showWarningDialog("<%=Encode.forJavaScriptAttribute(userFields[i].getDisplayName())%>"
```
Comment (malithie) [inline components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/edit.jsp]: encode for html and then encode for javascript

File: components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/edit.jsp [hunk 6]
```diff
@@ -163,7 +164,8 @@
                             var reg = new RegExp("<%=userFields[i].getRegEx()%>");
```
Comment (malithie) [inline components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/edit.jsp]: encode for javascript here

File: components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/edit.jsp [hunk 7]
```diff
@@ -163,7 +164,8 @@
...
                             var valid = reg.test(value);
                             if (value != '' && !valid) {
-                                CARBON.showWarningDialog("<%=CharacterEncoder.getSafeText(userFields[i].getDisplayName())%>" + " <fmt:message key='is.not.valid'/>");
+                                CARBON.showWarningDialog("<%=Encode.forJavaScriptAttribute(userFields[i].getDisplayName())%>"
```
Comment (malithie) [inline components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/edit.jsp]: encode for html and then encode for javascript

File: components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/index.jsp [hunk 8]
```diff
@@ -113,8 +113,9 @@
             if("true".equals(editCancel) && "true".equals(fromUserMgt)){
                 forwardTo = "../user/user-mgt.jsp?ordinal=1";
             } else {
-                forwardTo = "edit.jsp?username=" + URLEncoder.encode(username) + "&profile=" + profiles[0].getProfileName()
-                            + "&fromUserMgt="+fromUserMgt+"&noOfProfiles=1";
+                forwardTo = "edit.jsp?username=" + Encode.forUriComponent(username) + "&profile=" +
+                        Encode.forUriComponent(profiles[0].getProfileName()) + "&fromUserMgt="+
+                        Encode.forUriComponent(fromUserMgt)+"&noOfProfiles=1";
```
Comment (malithie) [inline components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/index.jsp]: apply formatting

File: components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/index.jsp [hunk 9]
```diff
@@ -178,14 +179,15 @@
                  }
         	     CARBON.showConfirmationDialog("<fmt:message key='remove.message1'/>"+ profile +"<fmt:message key='remove.message2'/>",
                     function() {
-              	       location.href ="remove-profile.jsp?username="+URLEncoder.encode(username)+"&profile="+profile+"&fromUserMgt=<%=fromUserMgt%>";
+              	       location.href ="remove-profile.jsp?username="+username+"&profile="+profile+
+                               "&fromUserMgt=<%=Encode.forJavaScriptBlock(Encode.forUriComponent(fromUserMgt))%>";
```
Comment (malithie) [inline components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/index.jsp]: Simply encode username, profile and fromUserMgt variables here for uri. Also apply formatting

File: components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/set-finish.jsp [hunk 10]
```diff
@@ -40,9 +39,9 @@
 <%
     String BUNDLE = "org.wso2.carbon.identity.user.profile.ui.i18n.Resources";
     ResourceBundle resourceBundle = ResourceBundle.getBundle(BUNDLE, request.getLocale());
-    String username = CharacterEncoder.getSafeText(request.getParameter("username"));
-	String profile = CharacterEncoder.getSafeText(request.getParameter("profile"));
-    String profileConfiguration = CharacterEncoder.getSafeText(request.getParameter("profileConfiguration"));
+    String username = request.getParameter("username");
+	String profile = request.getParameter("profile");
```
Comment (malithie) [inline components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/set-finish.jsp]: apply formatting

**Comment by malithie** [comment]
Apply appropriate encoding for below lines.
https://github.com/wso2/carbon-identity/pull/1008/files#diff-232a52c15bf09df9cac5952843e88080R104
https://github.com/wso2/carbon-identity/pull/1008/files#diff-232a52c15bf09df9cac5952843e88080R112

File: components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/index.jsp (hunk 9 above)
Comment (ChamaraPhilipsuom) [inline components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/index.jsp]: username and profile are both java script variables. Can we encode them?

Comment (malithie) [inline components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/index.jsp]: Sorry didn't note that. Yes, we don't have a way to encode javascript variables at the moment. This line is ok. Apply formatting though

File: components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/index.jsp (hunk 8 above)
Comment (ChamaraPhilipsuom) [inline components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/index.jsp]: i applied formatting. but it doesn't change.

Comment (malithie) [inline components/user-mgt/org.wso2.carbon.identity.user.profile.ui/src/main/resources/web/userprofile/index.jsp]: Fixed with https://github.com/wso2/carbon-identity/pull/1063

**Pull request merged**